internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -73,32 +73,57 @@ func (cw *CloudWatch) AlarmsWithTags(tags map[string]string) ([]AlarmStatus, err
 		}
 		alarmNames = append(alarmNames, name)
 	}
-	return cw.AlarmStatus(alarmNames)
+	return cw.AlarmStatuses(WithNames(alarmNames))
 }
 
-// AlarmStatus returns the status of each given alarm name.
-func (cw *CloudWatch) AlarmStatus(alarms []string) ([]AlarmStatus, error) {
-	if len(alarms) == 0 {
+// DescribeAlarmOpts sets the optional parameter for DescribeAlarms
+type DescribeAlarmOpts func(input *cloudwatch.DescribeAlarmsInput)
+
+// WithNames sets DescribeAlarms to filter on alarm names.
+func WithNames(names []string) DescribeAlarmOpts {
+	return func(in *cloudwatch.DescribeAlarmsInput) {
+		in.AlarmNames = aws.StringSlice(names)
+	}
+}
+
+// WithPrefix sets DescribeAlarms to filter on a name prefix.
+func WithPrefix(prefix string) DescribeAlarmOpts {
+	return func(in *cloudwatch.DescribeAlarmsInput) {
+		in.AlarmNamePrefix = aws.String(prefix)
+	}
+}
+
+// AlarmStatuses returns the status of each given alarm name.
+func (cw *CloudWatch) AlarmStatuses(opts ...DescribeAlarmOpts) ([]AlarmStatus, error) {
+	if len(opts) == 0 {
 		return nil, nil
 	}
-	var alarmStatus []AlarmStatus
+	var alarmStatuses []AlarmStatus
 	var err error
 	alarmResp := &cloudwatch.DescribeAlarmsOutput{}
+	in := &cloudwatch.DescribeAlarmsInput{
+		NextToken: alarmResp.NextToken,
```

Comment:
Why do we need to assign `NextToken` here?

Reply:
I had added it at some point bc it seemed necessary ... but whatever behavior it was preventing is no longer happening without it! Deleted.